src/sql/workbench/browser/scriptingUtils.ts

```diff
@@ -132,61 +113,51 @@ export function GetScriptOperationName(operation: ScriptOperation) {
 /**
  * Script the object as a statement based on the provided action (except Select)
  */
-export function script(connectionProfile: IConnectionProfile, metadata: azdata.ObjectMetadata,
+export async function script(connectionProfile: IConnectionProfile, metadata: azdata.ObjectMetadata,
 	connectionService: IConnectionManagementService,
 	queryEditorService: IQueryEditorService,
 	scriptingService: IScriptingService,
 	operation: ScriptOperation,
-	errorMessageService: IErrorMessageService): Promise<void> {
-	return new Promise<void>((resolve, reject) => {
-		connectionService.connectIfNotConnected(connectionProfile).then(connectionResult => {
-			let paramDetails = getScriptingParamDetails(connectionService, connectionResult, metadata);
-			scriptingService.script(connectionResult, metadata, operation, paramDetails).then(result => {
-				if (result) {
-					let script: string = result.script;
-
-					if (script) {
-						let description = (metadata.schema && metadata.schema !== '') ? `${metadata.schema}.${metadata.name}` : metadata.name;
-						queryEditorService.newSqlEditor(script, connectionProfile.providerName, undefined, description).then((owner) => {
-							// Connect our editor to the input connection
-							let options: IConnectionCompletionOptions = {
-								params: { connectionType: ConnectionType.editor, runQueryOnCompletion: RunQueryOnConnectionMode.none, input: owner },
-								saveTheConnection: false,
-								showDashboard: false,
-								showConnectionDialogOnError: true,
-								showFirewallRuleOnError: true
-							};
-							connectionService.connect(connectionProfile, owner.uri, options).then(() => {
-								resolve();
-							});
-						}).catch(editorError => {
-							reject(editorError);
-						});
-					} else {
-						let scriptNotFoundMsg = nls.localize('scriptNotFoundForObject', "No script was returned when scripting as {0} on object {1}",
-							GetScriptOperationName(operation), metadata.metadataTypeName);
-						let messageDetail = '';
-						let operationResult = scriptingService.getOperationFailedResult(result.operationId);
-						if (operationResult && operationResult.hasError && operationResult.errorMessage) {
-							scriptNotFoundMsg = operationResult.errorMessage;
-							messageDetail = operationResult.errorDetails;
-						}
-						if (errorMessageService) {
-							let title = nls.localize('scriptingFailed', "Scripting Failed");
-							errorMessageService.showDialog(Severity.Error, title, scriptNotFoundMsg, messageDetail);
-						}
-						reject(scriptNotFoundMsg);
-					}
-				} else {
-					reject(nls.localize('scriptNotFound', "No script was returned when scripting as {0}", GetScriptOperationName(operation)));
-				}
-			}, scriptingError => {
-				reject(scriptingError);
-			});
-		}).catch(connectionError => {
-			reject(connectionError);
-		});
-	});
+	errorMessageService: IErrorMessageService): Promise<boolean> {
```

Comment:
Why are we changing the return type.

Reply:
We weren't checking if the script actually connects. And when this gets consumed, the function that consumes it returns a boolean.